Refactor Intake per Driver Request

FireShredder24 opened the pull request:
This refactors the IntakeSubsystem's roller control to match the requested control scheme submitted by this year's drive team: Have the rollers spin at full speed in the direction of rejection until a button is pressed.

FireShredder24 committed:
Refactor Intake Roller Ctrl

To accomplish the functionality desired by the drivers

FireShredder24 committed:
Cleanup stale bools

terbos90803 commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -90,9 +90,7 @@ public RobotContainer() {
-    intakeRollerFwdButton.whenPressed(new InstantCommand(intakeSubsystem::runIntake, intakeSubsystem))
-        .whenReleased(new InstantCommand(intakeSubsystem::stopIntake, intakeSubsystem));
-    intakeRollerFwdRevToggle.whenPressed(new InstantCommand(intakeSubsystem::reverseIntake, intakeSubsystem));
+    intakeRollerFwdButton.whenPressed(new InstantCommand(intakeSubsystem::runIntake, intakeSubsystem));
```
You probably want this to be .whileHeld

FireShredder24 committed:
Impl Acceleration Limit

To prevent motor damage by switching back and forth too quickly

FireShredder24 committed:
Increase Intake Accel Ramp Speed

terbos90803 commented on src/main/java/frc/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -22,6 +22,8 @@
       Constants.INTAKE_PISTON_PCM_CHAN_RT_STOW);
 
   public boolean isLowered = false;
+  private double newPower;
```
newPower should not be a class member.  It should be a local.

terbos90803 commented on src/main/java/frc/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -22,6 +22,8 @@
 
   public boolean isLowered = false;
+  private double newPower;
+  private double oldPower;
```
A more typical name for this type of thing is "lastPower".

FireShredder24 committed:
Ensure Intake Runs WhileHeld

Previously it would only decrease its velocity briefly before going back to full rejection

terbos90803 commented on src/main/java/frc/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -30,16 +32,18 @@ public IntakeSubsystem() {
-      intakeSP.set(-MoPrefs.getIntakeRollerSetpoint());
-    else
+    if (isLowered) {
+      newPower = Math.max(oldPower - 0.04, -MoPrefs.getIntakeRollerSetpoint());
```
0.04 should be a MoPref.
Default will likely be closer to 0.1

terbos90803 commented on src/main/java/frc/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -30,16 +32,18 @@ public IntakeSubsystem() {
-      intakeSP.set(MoPrefs.getIntakeRollerSetpoint());
-    else
+    if (isLowered) {
+      newPower = Math.min(oldPower + 0.04, MoPrefs.getIntakeRollerSetpoint());
```
Same as above.

FireShredder24 committed:
Make Accel Ramp a MoPref

FireShredder24 committed:
Merge branch 'master' into IntakeRefactor

FireShredder24 committed:
Fix merge error

terbos90803 committed:
Cleanup merge
